core/src/main/java/org/apache/calcite/util/trace/CalciteTrace.java

```diff
@@ -118,6 +118,11 @@ public static Logger getSqlToRelTracer() {
     return LoggerFactory.getLogger("org.apache.calcite.sql2rel");
   }
 
+  public static Logger getRuleAttemptsTracer() {
+    return LoggerFactory.getLogger(
+        "org.apache.calcite.plan.AbstractRelOptPlanner.RuleAttemptsListener");
```

Comment:
Better use `RuleAttemptsListener.class.getName()`?

Reply:
Agree with @hsyuan

Reply:
`RuleAttemptsListener` is private, so there's no way to use it here.

What if we use `AbstractRelOptPlanner.class.getName() + ".rule_execution_summary"` or something like that?

Reply:
I don't want to break from the convention that we use full class name as logger name. There are already a couple of hard coded class names there. I personally don't think it's a problem.

Reply:
done